### PR TITLE
Don't snapshot unrelated test outputs

### LINKED ISSUE #14732: New race condition testing not working correctly
## Bug Description

A recent change to CIVET appears to check for race conditions in testing.  I believe the command is `--pedantic-checks`.  It is observed that this check is causing failure in BISON but the reported error is nonsense and incorrect. The tests listed by the check absolutely do **not** modify the list of files that are being reported under the heading

` Each of the tests in this set create or modify each of the following files:`

From what I've observed it only occurs during the recover and heavy testing targets. 

## Steps to Reproduce

In the case of the recover failures I think the source of this incorrect behavior has to do with the directory structure.  Consider two tests files set up as so:

```
test/tests/my_awesome_model/tests
test/tests/my_awesome_model/more_specific_tests/tests
```

Upon recover the `--pedantic-checks` is reporting that input files insides of `my_awesome_model` are writing to files located in `my_awesome_model/more_specific_tests` which is not happening.  

In the heavy tests case it appears that the `--pedantic-checks` is not honoring any setting of `Outputs/file_base=` inside of `cli_args` in the `tests` file as it says all files are writing to a file that is not actually created by any of the tests due to the setting of `Outputs/file_base=`.  

Moreover, the reporting of the tests that fail the check appears to be sporadic.  Each time the tests are run a different set of inputs are being reported as having race conditions in CIVET. 

## Impact

Significant impact.  Cannot do any development as all MRs are failing on BISON.

Can new CIVET recipes and checks not be turned on and tested prior to roll out to all applications causing failures?

### PR DESCRIPTION
**Pull request by permcody**
closes #14732

### CODE REVIEW

**Review by milljm — DISMISSED** [review]
Looks good!

**Comment by lindsayad** [comment]
@permcody You should rebase this now that #14731 is merged

**Comment by permcody** [comment]
Just did

**Review by lindsayad — DISMISSED** [review]

**Comment by permcody** [comment]
Damn - looks like I need to modify the "dirnames" in-place... hang on

File: python/TestHarness/FileChecker.py [hunk 1]
```diff
@@ -27,6 +27,12 @@ def getNewTimes(self):
     def get_all_files(self, job, times):
         """ Method to get the names and last_modified_times of all files within current test location """
         for dirpath, dirnames, filenames in os.walk(job.getTestDir(), followlinks=True):
+
+            # When perfoming a snapshot, don't traverse into directories that have test
+            # specifications of their own
+            if os.path.exists(os.path.join(dirpath, self.__input_file_name)):
+                dirnames[:] = []
```
Comment (permcody) [inline python/TestHarness/FileChecker.py]: That `[:]` is required!

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**